commonknowledge/mapped
# Validate phone number when importing external data [MAP-631]

## Description
This PR validates phone numbers when an external data source is imported or updated
If phone number is not a valid UK number it imports/updates the value as None

## Motivation and Context
Addresses issue [MAP-631](https://linear.app/commonknowledge/issue/MAP-631/disable-sms-button-if-number-is-not-valid)

## How Can It Be Tested?

1. Run the branch locally
2. Import a data source with a mixture of valid and invalid values in phone number field
3. Select People for Type of Data Source
4. Make sure to map the phone number field in the data source with Phone Field in Mapped
5. Load the data source onto a Map
6. Observe that only valid phone numbers have a call or SMS option
7. If you like add `console.log(selectedPointData, 'selected point data')` to the MarkerPopup component and observe the way the data is changed

![Untitled](https://github.com/user-attachments/assets/129ed0e2-af4a-4700-a12e-36232b965760)
